### ISSUE TITLE
Recursive anchors generates a stack overflow

### ISSUE THREAD

**Issue by DarkaMaul** [rhysd/actionlint#610]
Recursive anchors support is non consistent in the project.

The following workflow (from the documentation) is correctly parsed by `actionlint` ( [here](https://rhysd.github.io/actionlint/#eNpsj8FqwzAQRO/+ijkUHwp27/qZothLomKvxI7kFEL+vYjGwiE5mRm/p92N6pAKL133E090HZCFuX4Bim1hEv4nQM9Bf/cAhNWfxT3axVev/ZtMZtEc/EKH/pAaARSKqV/F4eN2A2UyyRxri/v9wCVPXqPNz9ze7qwV5VCvKaeiuQyHhZgltSOGSjqMX3O86hL9PPLSholuDp+v6zappLdK/07pTaZiDJs0+PEK4yrfnCyk/Dq9WX8BAAD//wHceU8=))
```yaml
on: push

jobs:
  test:
    services:
      nginx:
        image: nginx:latest
        credentials: &credentials
          username: ${{ secrets.user }}
          password: ${{ secrets.password }}
    runs-on: ubuntu-latest
    steps:
      - run: ./download.sh
        env: *credentials
      - run: ./upload.sh
        env: &credentials
      - &recursive
        run: ./some_script.sh
        env: *recursive
```

However, another instance of recursive anchors:

```yaml
# Reproducer: Anchor causes crash
name: Anchor Recursive

on: push

jobs:
  job1:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        include: &recursive_include
          - os: ubuntu-latest
            nested: *recursive_include
    steps:
      - run: echo "recursive test"%
```

Will generate the following stackoverflow:

```console
❯ actionlint ./.github/workflows/anchor-recursive.yml
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0x14020200450 stack=[0x14020200000, 0x14040200000]
fatal error: stack overflow

runtime stack:
runtime.throw({0x100b87744?, 0x10096c610?})
        runtime/panic.go:1094 +0x34 fp=0x16fa36da0 sp=0x16fa36d70 pc=0x1009af0d4
runtime.newstack()
        runtime/stack.go:1159 +0x44c fp=0x16fa36ed0 sp=0x16fa36da0 pc=0x100996a0c
runtime.morestack()
        runtime/asm_arm64.s:392 +0x70 fp=0x16fa36ed0 sp=0x16fa36ed0 pc=0x1009b4290

goroutine 1 gp=0x140000021c0 m=10 mp=0x14000980008 [running]:
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue.(*parser).parseMappingAt.(*parser).parseMapping.func1(0x14020200558)
        github.com/rhysd/actionlint/parse.go:313 +0x498 fp=0x14020200450 sp=0x14020200450 pc=0x100b584e8
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue(0x140401ff5b8, 0x1400007f7c0)
        github.com/rhysd/actionlint/parse.go:872 +0x1cc fp=0x14020200590 sp=0x14020200450 pc=0x100b57e4c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue(0x140401ff5b8, 0x1400007f720)
        github.com/rhysd/actionlint/parse.go:865 +0x29c fp=0x140202006d0 sp=0x14020200590 pc=0x100b57f1c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue-range1({{0x1400001090a?, 0x1400007f9a0?}, 0x1400c19db20?, 0x1400007f720?})
        github.com/rhysd/actionlint/parse.go:873 +0x5c fp=0x14020200730 sp=0x140202006d0 pc=0x100b5854c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue.(*parser).parseMappingAt.(*parser).parseMapping.func1(0x14020200a68)
        github.com/rhysd/actionlint/parse.go:355 +0x2d4 fp=0x14020200960 sp=0x14020200730 pc=0x100b58324
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue(0x140401ff5b8, 0x1400007f7c0)
        github.com/rhysd/actionlint/parse.go:872 +0x1cc fp=0x14020200aa0 sp=0x14020200960 pc=0x100b57e4c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue(0x140401ff5b8, 0x1400007f720)
        github.com/rhysd/actionlint/parse.go:865 +0x29c fp=0x14020200be0 sp=0x14020200aa0 pc=0x100b57f1c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue-range1({{0x1400001090a?, 0x1400007f9a0?}, 0x1400c19dae0?, 0x1400007f720?})
        github.com/rhysd/actionlint/parse.go:873 +0x5c fp=0x14020200c40 sp=0x14020200be0 pc=0x100b5854c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue.(*parser).parseMappingAt.(*parser).parseMapping.func1(0x14020200f78)
        github.com/rhysd/actionlint/parse.go:355 +0x2d4 fp=0x14020200e70 sp=0x14020200c40 pc=0x100b58324
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue(0x140401ff5b8, 0x1400007f7c0)
        github.com/rhysd/actionlint/parse.go:872 +0x1cc fp=0x14020200fb0 sp=0x14020200e70 pc=0x100b57e4c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue(0x140401ff5b8, 0x1400007f720)
        github.com/rhysd/actionlint/parse.go:865 +0x29c fp=0x140202010f0 sp=0x14020200fb0 pc=0x100b57f1c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue-range1({{0x1400001090a?, 0x1400007f9a0?}, 0x1400c19daa0?, 0x1400007f720?})
        github.com/rhysd/actionlint/parse.go:873 +0x5c fp=0x14020201150 sp=0x140202010f0 pc=0x100b5854c
github.com/rhysd/actionlint.(*parser).parseRawYAMLValue.(*parser).parseMappingAt.(*parser).parseMapping.func1(0x14020201488)
        github.com/rhysd/actionlint/parse.go:355 +0x2d4 fp=0x14020201380 sp=0x14020201150 pc=0x100b58324
```

Of note, both workflows are invalid and rejected by the GitHub action parser anyway. So I'm not sure if it's really a priority to fix this.

# Setup
- actionlint installed via homebrew:
```console
❯ actionlint --version
1.7.10
installed by building from source
built with go1.25.5 compiler for darwin/arm64
```
- Running on macOS Tahoe (ARM)

**Comment by rhysd** [comment]
Thank you for reporting this. I could reproduce it.
